### ISSUE TITLE
Add syntax `A[i=>1,j=>2]` as syntax for getting/setting elements of an ITensor

### ISSUE THREAD

**Issue by mtfishman** [ITensor/ITensors.jl#106]
This should just involve making a new `IndexVal` constructor that takes a `Pair{Index,Int}`, then extending the definition of `getindex(::ITensor,...)`

**Comment by mtfishman** [comment]
Something I just noticed while randomly playing around with the REPL:

```julia
julia> =>(i::Int,j::Int) = i+j
=> (generic function with 1 method)

julia> 1=>2
3
```

so we can directly overload the `=>` operator. This could give us a simpler way to implement this, since we could just define:

```julia
=>(i::Index,val::Int) = IndexVal(i,val)
```

and avoid having to use `Pair` anywhere. The only downside I see is that it removes that notation if you actually _want_ a `Pair{Index,Int}`, but you could always just call the `Pair` constructor directly.

**Comment by emstoudenmire** [comment]
I think this is great for most, if not all purposes. It's definitely nice in that it's a minimal amount of extra code to get a great feature. And there's something to be said for not having more types flying around. (Meaning that we don't have to explain that `Pair{Index,Int}` means the same thing as an `IndexVal`.)

The only downside I can see is that we are sort of breaking, or hijacking a feature of the language. Meaning that if someone *did* expect i=>2 to give a Pair and then it didn't, they could be surprised and possibly spend too long trying to figure out what happened. 

So I currently still vote we do it this way, but let's think about if that downside I just mentioned is acceptable.

**Comment by mtfishman** [comment]
Closed by #227 (it was easy enough to do without defining `=>(i::Index,val::Int) = IndexVal(i,val)`, so I avoided that).
